JDK15 java/lang/StackWalker/CallerSensitiveMethod/Main.java failure - RuntimeException: Got class GetCallerClassTest$LambdaTest$$Lambda$3/0000000022A31F70, but expected class GetCallerClassTest$LambdaTest

Failure link
------------

https://ci.eclipse.org/openj9/job/Test_openjdk15_j9_sanity.openjdk_ppc64_aix_Nightly/26/consoleFull

Optional info
-------------


Failure output (captured from console output)
---------------------------------------------

```
01:42:41  TEST: java/lang/StackWalker/CallerSensitiveMethod/Main.java
01:42:41  TEST JDK: /home/jenkins/workspace/Test_openjdk15_j9_sanity.openjdk_ppc64_aix_Nightly/openjdkbinary/j2sdk-image
01:42:41  
01:42:43  STDERR:
01:42:43  Unexpected exception:
01:42:43  java.lang.RuntimeException: java.lang.RuntimeException: Got class GetCallerClassTest$LambdaTest$$Lambda$3/0000000022A31F70, but expected class GetCallerClassTest$LambdaTest
01:42:43  	at GetCallerClassTest$LambdaTest.lambda$run$0(GetCallerClassTest.java:197)
01:42:43  	at GetCallerClassTest$LambdaTest$$Lambda$3/0000000022A31F70.run(Unknown Source)
01:42:43  	at GetCallerClassTest$LambdaTest.run(GetCallerClassTest.java:200)
01:42:43  	at GetCallerClassTest.test(GetCallerClassTest.java:77)
01:42:43  	at GetCallerClassTest.main(GetCallerClassTest.java:70)
01:42:43  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
01:42:43  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
01:42:43  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
01:42:43  	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
01:42:43  	at com.sun.javatest.regtest.agent.MainWrapper$MainThread.run(MainWrapper.java:127)
01:42:43  	at java.base/java.lang.Thread.run(Thread.java:851)
01:42:43  Caused by: java.lang.RuntimeException: Got class GetCallerClassTest$LambdaTest$$Lambda$3/0000000022A31F70, but expected class GetCallerClassTest$LambdaTest
01:42:43  	at GetCallerClassTest.assertEquals(GetCallerClassTest.java:159)
01:42:43  	at GetCallerClassTest$LambdaTest.lambda$run$0(GetCallerClassTest.java:188)
01:42:43  	... 10 more

01:42:44  STDERR:
01:42:44  java.lang.RuntimeException: Hidden frames are shown
01:42:44  	at HiddenFrames.walk(HiddenFrames.java:74)
01:42:44  	at HiddenFrames.test(HiddenFrames.java:60)
01:42:44  	at HiddenFrames.main(HiddenFrames.java:40)
01:42:44  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
01:42:44  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
01:42:44  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
01:42:44  	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
01:42:44  	at com.sun.javatest.regtest.agent.MainActionHelper$AgentVMRunnable.run(MainActionHelper.java:298)
01:42:44  	at java.base/java.lang.Thread.run(Thread.java:851)
01:42:44  
01:42:44  JavaTest Message: Test threw exception: java.lang.RuntimeException
01:42:44  JavaTest Message: shutting down test
01:42:44  
01:42:44  
01:42:44  TEST RESULT: Failed. Execution failed: `main' threw exception: java.lang.RuntimeException: Hidden frames are shown

02:09:04  TEST: java/util/logging/Logger/getLogger/TestInferCaller.java
02:09:04  STDERR:
02:09:04  java.lang.RuntimeException: Bad class name: 
02:09:04  	expected: TestInferCaller
02:09:04  	actual:   TestInferCaller$$Lambda$44/0000000022A16420
02:09:04  	at TestInferCaller.assertEquals(TestInferCaller.java:373)
02:09:04  	at TestInferCaller.checkEvent(TestInferCaller.java:380)
02:09:04  	at TestInferCaller.testLambda(TestInferCaller.java:328)
02:09:04  	at TestInferCaller.test5(TestInferCaller.java:313)
02:09:04  	at TestInferCaller.test(TestInferCaller.java:345)
02:09:04  	at TestInferCaller.main(TestInferCaller.java:359)
02:09:04  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
02:09:04  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
02:09:04  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
02:09:04  	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
02:09:04  	at com.sun.javatest.regtest.agent.MainWrapper$MainThread.run(MainWrapper.java:127)
02:09:04  	at java.base/java.lang.Thread.run(Thread.java:851)
02:09:04  
02:09:04  JavaTest Message: Test threw exception: java.lang.RuntimeException: Bad class name: 
02:09:04  	expected: TestInferCaller
02:09:04  	actual:   TestInferCaller$$Lambda$44/0000000022A16420
02:09:04  JavaTest Message: shutting down test
02:09:04  
02:09:04  STATUS:Failed.`main' threw exception: java.lang.RuntimeException: Bad class name: expected: TestInferCaller actual: TestInferCaller$$Lambda$44/0000000022A16420
```

For example, to rebuild the failed tests in <Jenkins URL>=https://ci.eclipse.org/openj9/job/Grinder, use the following links:
02:43:27  https://ci.eclipse.org/openj9/job/Grinder/parambuild/?JDK_VERSION=15&JDK_IMPL=openj9&BUILD_LIST=openjdk&PLATFORM=ppc64_aix&TARGET=jdk_lang_j9_0

This could be related to hidden class changes.

fyi @hangshao0

--- a/runtime/oti/VMHelpers.hpp
+++ b/runtime/oti/VMHelpers.hpp
@@ -1529,7 +1529,9 @@ exit:
 	{
 		bool isHidden = false;
 		J9Class* currentClass = J9_CLASS_FROM_METHOD(method);
-		if (J9_ARE_ANY_BITS_SET(currentClass->classFlags, J9ClassIsAnonymous)) {
+		if (J9_ARE_ANY_BITS_SET(currentClass->classFlags, J9ClassIsAnonymous)
+			|| J9ROMCLASS_IS_HIDDEN(currentClass->romClass)
+		) {
 			/* lambda helper method */
 			isHidden = true;
 		} else {

